proposals/2024-09-01-Tags-Improvement.md

```diff
@@ -0,0 +1,154 @@
+# Tags Improvement
+
+
+## Metadata
+
+|Tag |Value |
+|---- | ---------------- |
+|Proposal |[2024-09-01-Tags-Improvement](https://github.com/OAI/OpenAPI-Specification/tree/main/proposals/{2024-09-01-Tags-Improvement.md})|
+|Authors|[Lorna Mitchell](https://github.com/lornajane)|
+|Review Manager | TBD |
+|Status |Proposal|
+|Implementations | |
+|Issues | [1367](https://github.com/OAI/OpenAPI-Specification/issues/1367), [{Iss}](https://github.com/OAI/OpenAPI-Specification/issues/{Iss}), |
+|Previous Revisions | None, but see [Moonwalk discussion](https://github.com/OAI/sig-moonwalk/discussions/67)
+
+## Change Log
+
+|Date |Responsible Party |Description |
+|---- | ---------------- | ---------- |
+| 2024-09-01 | @lornajane | Initial draft |
+
+## Introduction
+
+Evolve the existing `tags` implementation to (optionally) support more use cases, such as giving the tags some grouping/relationships and adding more metadata.
+
+## Motivation
+
+The tags feature hasn't changed since the spec was still called Swagger, but it's one of the most-extended aspects of OpenAPI with most tools supporting some sort of grouping or additional metadata. One motivation for this proposal is to "pave the cowpath" and formalise some of the patterns from those extensions as part of the specification.
+
+The existing tags implementation is also quite limiting, so users/tools feel they "cannot" use tags, because they are so widely implemented for documentation navigation and required exactly one tag per operation, to be used only for this single purpose or to be opted out using extensions such as `x-traitTag`. The result is that we see lots of extensions that add arbitrary metadata to operations, some of them even become part of the specification officially (looking at you, `deprecated: true`) or being so widely used that we should probably adopt them (`x-internal`, `x-experimental`). The specification does have a way of tagging operations, but it's too limited and the result is that everything has to be added as an extension field.
```

Comment:
"required exactly one tag per operation" - is that an example for (ab)use of tags?

Reply:
It's a very common pattern of what downstream tools expect

Reply:
Well...